### PR TITLE
New Resource: azurerm_stream_analytics_function_javascript_uda

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/7303

--- PASS: TestAccStreamAnalyticsFunctionJavaScriptUDA_basic (244.77s)
--- PASS: TestAccStreamAnalyticsFunctionJavaScriptUDA_requiresImport (282.79s)
--- PASS: TestAccStreamAnalyticsFunctionJavaScriptUDA_update (288.96s)
--- PASS: TestAccStreamAnalyticsFunctionJavaScriptUDA_inputs (464.66s)

### CODE REVIEW

File: website/docs/r/stream_analytics_function_javascript_uda.html.markdown [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+A `input` block supports the following:
+
+* `type` - The Data Type for the Input Argument of this JavaScript Function. Possible values include `any`, `array`, `bigint`, `datetime`, `float`, `nvarchar(max)` and `record`.
+
+---
+
+A `output` block supports the following:
+
+* `type` - The Data Type output from this JavaScript Function. Possible values include `any`, `array`, `bigint`, `datetime`, `float`, `nvarchar(max)` and `record`.
```
Comment (stephybun) [inline website/docs/r/stream_analytics_function_javascript_uda.html.markdown:82]:
```suggestion
An `input` block supports the following:

* `type` - The input data type of this JavaScript Function. Possible values include `any`, `array`, `bigint`, `datetime`, `float`, `nvarchar(max)` and `record`.

---

An `output` block supports the following:

* `type` - The output data type from this JavaScript Function. Possible values include `any`, `array`, `bigint`, `datetime`, `float`, `nvarchar(max)` and `record`.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @neil-yechenwei, this is off to a great start! I left some suggestions in line, once those are fixed up we can take another look.

File: website/docs/r/stream_analytics_function_javascript_uda.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/stream_analytics_function_javascript_uda.html.markdown:82]: Updated

File: website/docs/r/stream_analytics_function_javascript_uda.html.markdown [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+* `name` - (Required) The name of the JavaScript UDA Function. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Stream Analytics Job exists. Changing this forces a new resource to be created.
+
+* `stream_analytics_job_id` - (Required) The resource ID of the Stream Analytics Job where this Function should be created. Changing this forces a new resource to be created.
+
+* `input` - (Required) One or more `input` blocks as defined below.
+
+* `output` - (Required) An `output` blocks as defined below.
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply website/docs/r/stream_analytics_function_javascript_uda.html.markdown]: Updated

File: website/docs/r/stream_analytics_function_javascript_uda.html.markdown [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+subcategory: "Stream Analytics"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_stream_analytics_function_javascript_uda"
+description: |-
+  Manages a JavaScript UDA Function within Stream Analytics Streaming Job.
+---
+
+# azurerm_stream_analytics_function_javascript_uda
+
+Manages a JavaScript UDA Function within Stream Analytics Streaming Job.
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply website/docs/r/stream_analytics_function_javascript_uda.html.markdown]: Updated

File: website/docs/r/stream_analytics_function_javascript_uda.html.markdown [hunk 4]
```diff
@@ -0,0 +1,105 @@
+---
+subcategory: "Stream Analytics"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_stream_analytics_function_javascript_uda"
+description: |-
+  Manages a JavaScript UDA Function within Stream Analytics Streaming Job.
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply website/docs/r/stream_analytics_function_javascript_uda.html.markdown]: Updated

File: internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+	defer cancel()
+
+	id, err := parse.FunctionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if resp, err := client.Delete(ctx, id.ResourceGroup, id.StreamingjobName, id.Name); err != nil {
+		if !response.WasNotFound(resp.Response) {
+			return fmt.Errorf("deleting %s: %+v", id, err)
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go]: Updated

File: internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+	id, err := parse.FunctionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.StreamingjobName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] %q was not found - removing from state!", id)
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go]: Updated

File: internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", id, err)
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go]: Updated

File: internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+			},
+
+			"stream_analytics_job_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.StreamingJobID,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go]: Yes. You're right. Removed it.

File: internal/services/streamanalytics/registration.go [hunk 9]
```diff
@@ -53,6 +53,7 @@ func (r Registration) SupportedResources() map[string]*pluginsdk.Resource {
 	return map[string]*pluginsdk.Resource{
 		"azurerm_stream_analytics_job":                     resourceStreamAnalyticsJob(),
 		"azurerm_stream_analytics_function_javascript_udf": resourceStreamAnalyticsFunctionUDF(),
+		"azurerm_stream_analytics_function_javascript_uda": resourceStreamAnalyticsFunctionUDA(),
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/streamanalytics/registration.go]: Updated

File: internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceStreamAnalyticsFunctionUDA() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceStreamAnalyticsFunctionUDACreateUpdate,
+		Read:   resourceStreamAnalyticsFunctionUDARead,
+		Update: resourceStreamAnalyticsFunctionUDACreateUpdate,
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/streamanalytics/stream_analytics_function_javascript_uda_resource.go]: Updated

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for your comments. I've updated code. Please take another look. Thanks.

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei, LGTM 🚀

**Pull request merged**